src/Console/ConfigurationResolver.php

```diff
@@ -528,6 +536,19 @@ public function getUnsupportedPhpVersionAllowed(): bool
         return $this->isUnsupportedPhpVersionAllowed;
     }
 
+    public function getFilterFixerByFile(): ?\Closure
+    {
+        if (false === $this->isFilterFixerByFileResolved) {
+            $config = $this->getConfig();
+            $this->filterFixerByFile = $config instanceof FilterFixerByFileAwareConfigInterface
+                ? $config->getFilterFixerByFile()
+                : null;
+            $this->isFilterFixerByFileResolved = true;
+        }
+
+        return $this->filterFixerByFile;
```

Comment:
Do we really need this level of complexity here? Wouldn't it be enough?


```suggestion
        return $config instanceof FilterFixerByFileAwareConfigInterface
            ? $config->getFilterFixerByFile()
            : null;
```

Since it's a `Closure`, there's nothing to resolve because it work's more like a proxied getter. Calling this multiple times wouldn't be a problem.

Reply:
I kept the same approach as most of the other methods in that class. Why? Because in the future we may add support for specifying the filter via CLI options (maybe as the name of a class implementing `RuleCustomizationPolicyInterface`)